pkg/dogstatsd/enrich_bench_test.go

```diff
@@ -0,0 +1,45 @@
+package dogstatsd
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/DataDog/datadog-agent/pkg/tagger/collectors"
+)
+
+func buildTags(tagCount int) []string {
+	tags := make([]string, 0, tagCount)
+	for i := 0; i < tagCount; i++ {
+		tags = append(tags, fmt.Sprintf("tag%d:val%d", i, i))
+	}
+
+	return tags
+}
+
+// used to store the result and avoid optimizations
+var tags []string
+
+func BenchmarkEnrichTags(b *testing.B) {
+	originalGetTags := getTags
+
+	for i := 100; i < 1000; i *= 2 {
+		b.Run(fmt.Sprintf("%d-tags", i), func(sb *testing.B) {
+			baseTags := buildTags(i)
+			extraTags := buildTags(i / 2)
```

Comment:
I wonder if the alloc improvement you're seeing in your benchmark isn't simply a result of the dividing value of `2` you've chosen here.
Typically, if the `extraTags` slice always had a length of `1`, I think the results would be worse (since the new code would always allocate a new slice of cap `i+1`, whereas the old code would in most cases not do any allocation).

That said, in most cases, we'd expect the `extraTags` to be longer than `baseTags` right? (when origin tagging is enabled, I'd expect that there are more tags from origin tagging than actual tags on the ingested dogstatsd metrics). If so, then the new code is more efficient indeed. WDYT?

Reply:
Yes I think you're right, even though I designed the benchmark before changing the code, the benchmark is not optimal in terms of practical use cases.
I've changed it to be closer to reality and it does not show any difference between previous and current code. I'm guessing that the first append creates a large enough slice if `extraTags` is large enough.

Though, with the change you suggested, I think we are now guaranteed to have the minimum number of allocations in any situation, while before it was depending on the size of the different slices.